plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelector.php

```diff
@@ -17,16 +17,313 @@ class AddToCartWithOptionsVariationSelector extends AbstractBlock {
+		);
+
+		$html .= '<option value="">' . esc_html__( 'Choose an option', 'woocommerce' ) . '</option>';
+		$html .= $this->get_variation_options_html( $product, $attribute_name, $options, $selected, taxonomy_exists( $attribute_name ) );
+		$html .= '</select>';
+
+		return $html;
+	}
+
+	/**
+	 * Get selected attribute value.
+	 *
+	 * @param WC_Product $product The product object.
+	 * @param string     $attribute_name Name of the attribute.
+	 * @return string Selected value
+	 */
+	private function get_selected_attribute_value( $product, $attribute_name ): string {
+		return $product->get_variation_default_attribute( $attribute_name );
+	}
+
+	/**
+	 * Get HTML for variation options.
+	 *
+	 * @param WC_Product $product The product object.
+	 * @param string     $attribute_name Name of the attribute.
+	 * @param array      $options Available options.
+	 * @param string     $selected Selected value.
+	 * @param bool       $is_taxonomy Whether this is a taxonomy-based attribute.
+	 * @return string Options HTML
+	 */
+	private function get_variation_options_html( $product, $attribute_name, $options, $selected, $is_taxonomy ): string {
+		if ( empty( $options ) ) {
+			return '';
+		}
+
+		$html  = '';
+		$items = $is_taxonomy
+			? wc_get_product_terms( $product->get_id(), $attribute_name, array( 'fields' => 'all' ) )
+			: $options;
+
+		foreach ( $items as $item ) {
+			$option_value = $is_taxonomy ? $item->slug : $item;
+			$option_label = $is_taxonomy ? $item->name : $item;
+
+			if ( ! $is_taxonomy || in_array( $option_value, $options, true ) ) {
+				$selected_attr = $is_taxonomy
+					? selected( sanitize_title( $selected ), $option_value, false )
+					: selected( $selected, $option_value, false );
+
+				/**
+				 * Filter the variation option name.
+				 *
+				 * @since 9.7.0
+				 *
+				 * @param string     $option_label    The option label.
+				 * @param WP_Term|string|null $item   Term object for taxonomies, option string for custom attributes.
+				 * @param string     $attribute_name  Name of the attribute.
+				 * @param WC_Product $product         Product object.
+				 */
+				$filtered_label = apply_filters(
+					'woocommerce_variation_option_name',
+					$option_label,
+					$is_taxonomy ? $item : null,
+					$attribute_name,
+					$product
+				);
+
+				$html .= sprintf(
+					'<option value="%s" %s>%s</option>',
+					esc_attr( $option_value ),
+					$selected_attr,
+					esc_html( $filtered_label )
+				);
+			}
+		}
+
+		return $html;
+	}
+
+	/**
+	 * Render variation form.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param array      $attributes Block attributes.
+	 * @return string Rendered form HTML
+	 */
+	private function render_variation_form( $product, $attributes ): string {
+		$variation_attributes = $product->get_variation_attributes();
+		if ( empty( $variation_attributes ) ) {
+			return '';
+		}
+
+		wp_enqueue_script( 'wc-add-to-cart-variation' );
```

Comment:
Can we move this line under

```PHP
		$variations = $this->get_variations_data( $product );
		if ( empty( $variations ) ) {
			return '';
		}
```

No need to enqueue the script if we have no variations, no?

Reply:
Done in [3c97fb4](https://github.com/woocommerce/woocommerce/pull/53947/commits/3c97fb48a3fe26809fb934151c2154fa4e3259e7).